Sources/Transifex/Cache.swift

```diff
@@ -23,22 +23,51 @@ public typealias TXLocaleStrings = [String: TXStringInfo]
+    ///
+    /// - Parameter string: The string to be checked
+    /// - Returns: True if the string contains a translation, False otherwise.
+    static func containsTranslation(_ string: String?) -> Bool {
+        guard let string = string else {
+            return false
+        }
+    
+        return string.count > 0
+    }
+}
+
 /// Overriding policy `TXStringOverrideFilterCache` decorator class so that any translations fed by
 /// the `update()` method of the `TXCache` protocol are updated using one of the following policies.
+///
+/// You can find an easy to remember table containing a number of cases and how each policy updates
+/// the cache below:
+///
+///```
+/// | Key | Cache | New  | All  | Using Translated Only | Untranslated Only |
+/// |-----|-------|------|------|-----------------------|-------------------|
+/// | 1   | 1     | None | None | 1                     | 1                 |
+/// | 2   | 1     | 2    | 2    | 2                     | 1                 |
+/// | 3   | 1     | ""   | ""   | 1                     | 1                 |
+/// | 4   | ""    | None | None | ""                    | ""                |
+/// | 5   | ""    | 2    | 2    | 2                     | 2                 |
+/// | 6   | None  | 2    | 2    | 2                     | 2                 |
+///```
 @objc
 public enum TXCacheOverridePolicy : Int {
     /// All of the cache entries are replaced with the new translations.
-    /// Empty cache entries from the new translations are filtered out.
     case overrideAll
-    /// All new translations are added to the cache, either updating existing translations or adding
-    /// new ones.
-    /// If a translation is not found in cache but exists in the new translations, it's not added.
-    /// If a translation is found in cache but doesn't exist in the new translations, it's left untouched.
-    /// Empty cache entries from the new translations are filtered out.
+    /// New translations update the cache (updating existing entries or adding new ones) and any cached
+    /// translations that aren't updated are left as-is.
+    /// If a cached translation exists and it's not an empty string and the value for the new translation is an
+    /// empty string, then the cached translation is left as -is.
     case overrideUsingTranslatedOnly
-    /// Only the translations not existing in the cache are updated.
-    /// If a translation is found in the cache but not in the new translations, it's left untouched.
-    /// Empty cache entries from the new translations are filtered out.
+    /// Only the translations that don't exist in the cache or cached translations that contain empty strings
+    /// are updated / added, otherwise they are left as-is.
+    /// If a cached translation exists and it's not an empty string and the value for the new translation is an
+    /// empty string, then the cached translation is left as -is.
     case overrideUntranslatedOnly
```

Comment:
As discussed offline, let's remove `overrideUntranslatedOnly` completely, as it doesn't seem to cover a plausible customer flow.

Reply:
Removed.